File: app/src/v1/routers/puestos.py
from flask import Blueprint, jsonify, request
from src.v1.db.conexion import conexion

puestos = Blueprint('puestos_blueprint', __name__)

@puestos.route('/consulta', methods=['GET'])
def consulta_puesto():
    try:
        opcion = request.args.get('opcion')
        idpuesto = request.args.get('idpuesto')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionespuestos({opcion},{idpuesto},'',0,0);"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

@puestos.route('/agregar', methods=['POST'])
def agregar_puesto():
    try:
        data = request.json
        opcion = data.get('opcion')
        idpuesto = data.get('idpuesto')
        descripcion = data.get('descripcion')
        empleadoalta = data.get('empleadoalta')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionespuestos({opcion},{idpuesto},'{descripcion}',{empleadoalta},0);"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

@puestos.route('/modificar', methods=['PUT'])
def modificar_puesto():
    try:
        data = request.json
        opcion = data.get('opcion')
        idpuesto = data.get('idpuesto')
        descripcion = data.get('descripcion')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionespuestos({opcion},{idpuesto},'{descripcion}',0,0);"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

@puestos.route('/baja', methods=['DELETE'])
def baja_puesto():
    try:
        data = request.json
        opcion = data.get('opcion')
        idpuesto = data.get('idpuesto')
        empleadobaja = data.get('empleadobaja')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionespuestos({opcion},{idpuesto},'',0,{empleadobaja});"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

File: WWW/PHP/Clases/Puestos.php
<?php

declare(strict_types=1);

final class Puestos
{

    private static string $apiUrl = "http://api:3000/puestos";

    private function __construct()
    {
    }

    private static function callApi(string $method, string $endpoint, ?array $data = null): mixed
    {
        $url = self::$apiUrl . $endpoint;
        $ch = curl_init($url);

        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);

        if ($data) {
            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
            curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json'));
        }

        $response = curl_exec($ch);
        curl_close($ch);

        return json_decode((string) $response, true);
    }

    public static function consultaPuestoActivo(int $opcion, int $idpuesto): mixed
    {
        $endpoint = "/consulta?opcion={$opcion}&idpuesto={$idpuesto}";
        return self::callApi('GET', $endpoint);
    }

    public static function agregarPuesto(int $opcion, int $idpuesto, string $descripcion, int $empleadoalta): mixed
    {
        $data = array(
            'opcion' => $opcion,
            'idpuesto' => $idpuesto,
            'descripcion' => $descripcion,
            'empleadoalta' => $empleadoalta
        );
        return self::callApi('POST', '/agregar', $data);
    }

    public static function modificarPuesto(int $opcion, int $idpuesto, string $descripcion): mixed
    {
        $data = array(
            'opcion' => $opcion,
            'idpuesto' => $idpuesto,
            'descripcion' => $descripcion
        );
        return self::callApi('PUT', '/modificar', $data);
    }

    public static function bajaPuesto(int $opcion, int $idpuesto, int $empleadobaja): mixed
    {
        $data = array(
            'opcion' => $opcion,
            'idpuesto' => $idpuesto,
            'empleadobaja' => $empleadobaja
        );
        return self::callApi('DELETE', '/baja', $data);
    }

}